msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase1561081.java

```diff
@@ -0,0 +1,198 @@
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
+
+        //extract the device id claim from the access token.
+        String deviceId = (String) IDToken.parseJWT(authResult.getAccessToken()).get("deviceid");
+
+        //this gets the deviceId from the Ui and matches it to the deviceID obtained from the AT
+        getDeviceIdFromMyAccount(deviceId, username, password, true);
+
+        //init the release candidate BrokerHost in order to run wpj Leave
+        BrokerHost brokerHost = new BrokerHost(BrokerHost.BROKER_HOST_APK_RC);
+        if(brokerHost.isInstalled()){
+            brokerHost.uninstall();
+        }
+
+        brokerHost.install();
+        //run wpj leave
+        brokerHost.wpjLeave();
+
+        final MsalAuthTestParams authTestParams_SecondTry = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.LOGIN)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+
+        //acquire the token interactively for a second time.
+         authResult = msalSdk.acquireTokenInteractive(authTestParams_SecondTry, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
+                        .prompt(PromptParameter.LOGIN)
+                        .loginHint(username)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .build();
+
+                new MicrosoftStsPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
+
+        //wait for two minutes to ensure the changes are made.
+        Thread.sleep(TimeUnit.MINUTES.toMinutes(2));
+
+        getDeviceIdFromMyAccount(deviceId, username, password, false);
+
+    }
+
+    /**
+     * Get's the device registration details from <code>MY_ACCOUNT_MICROSOFT_URL</code> service
+     * and confirms the deivce's registration / de-registration status.
+     *
+     * @param deviceID - The device Id obtained from performDeviceRegistration
+     * @param username - LabApi username
+     * @param password - LabApi password
+     * @param isDeviceRegistered - toggle to enable re-usability of this method across the two states
+     *                           this device maybe in, ie REGISTERED / UNREGISTERED.
+     * @throws Throwable
+     */
+    private void getDeviceIdFromMyAccount(String deviceID, String username, String password, boolean isDeviceRegistered) throws  Throwable{
```

Comment:
Should we make this a helper method (UIAutomationUtilities), so other test cases can also use this.

Reply:
We should

Reply:
We already have the get device ID in the BrokerHost class that does the same thing. This method just uses the slightly longer approach to get it and compares it against the deviceId in the token. We can achieve the same using the brokerHost.getdeviceId() method.